src/main/java/org/zalando/stups/tokens/AccessTokensBuilder.java

```diff
@@ -599,6 +604,8 @@ private boolean isFilesystemSecretsLayout() {
         try {
             return getCredentialsDir().list(forSuffix("-token-secret")).length > 0;
         } catch (Exception e) {
```

Comment:
Isn't this a bit too broad? Wouldn't `FileNotFoundException` be enough?

Reply:
No, `list()` can also throw an `SecurityException`. Furthermore, `getCredentialsDir()' throws an `IllegalStateException' when `CREDENTIALS_DIR` is not provided by user as an environment variable.
I guess that's the reason why it fails when people move from STUPS to K8s I guess. This was handled by Taupage transparently and nobody specifies it in the deployment-manifest?

Do you want me to make it more specific for these two Exceptions?